packages/nodes-base/credentials/Aws.credentials.ts

```diff
@@ -531,4 +749,306 @@ export class Aws implements ICredentialType {
 			method: 'POST',
 		},
 	};
+
+	private async getSecurityHeaders(credentials: AwsCredentialsType): Promise<{
+		accessKeyId: string;
+		secretAccessKey: string;
+		sessionToken?: string;
+	}> {
+		// For backward compatibility: if credentialType is not set, default to 'iam'
+		const credentialType =
+			credentials.credentialType || (credentials.assumeRole ? 'assumeRole' : 'iam');
+
+		if (credentialType === 'iam') {
+			// Use manually provided credentials
+			if (!credentials.accessKeyId || credentials.accessKeyId.trim() === '') {
+				throw new Error('Access Key ID is required for IAM credentials.');
+			}
+			if (!credentials.secretAccessKey || credentials.secretAccessKey.trim() === '') {
+				throw new Error('Secret Access Key is required for IAM credentials.');
+			}
+			return {
+				accessKeyId: `${credentials.accessKeyId}`.trim(),
+				secretAccessKey: `${credentials.secretAccessKey}`.trim(),
+				sessionToken: credentials.temporaryCredentials
+					? `${credentials.sessionToken}`.trim()
+					: undefined,
+			};
+		} else {
+			throw new Error('Invalid credential type. Expected IAM credentials for direct access.');
+		}
+	}
+
+	private async getSystemCredentials(): Promise<{
+		accessKeyId: string;
+		secretAccessKey: string;
+		sessionToken?: string;
+	} | null> {
+		// 1. Check for explicit environment variables
+		const accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+		const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+		const sessionToken = process.env.AWS_SESSION_TOKEN;
+
+		if (accessKeyId && secretAccessKey) {
+			return {
+				accessKeyId: accessKeyId.trim(),
+				secretAccessKey: secretAccessKey.trim(),
+				sessionToken: sessionToken?.trim(),
+			};
+		}
+
+		// 2. Check for AWS_PROFILE environment variable
```

Comment:
Why not use the builtin SDK chain? https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromnodeproviderchain (AKA, just don't pass any credential object at all to the SDK call)

Reply:
But the existing credential code is not doing anything that could be using the AWS SDK. Loading system credential and calling STS API are thing that the SDK is made for. `nodes-base` or other packages already depends on some `@aws-sdk/`. You're trading external dependency with homemade, barely tested, AWS code.

Reply:
@isra17 the changes I made to the AWS credential retrieves a new set of temporary credentials any time the credential is used (it runs sts.AssumeRole any time cred is used). It does not cache anything (unless n8n does something internally), so no there are no issues with credential expiration. I am using the code of this PR in our environment with no issues.